core/src/main/java/org/testcontainers/containers/ContainerState.java

```diff
@@ -78,16 +78,7 @@ default boolean isRunning() {
      * @return is the container created?
      */
     default boolean isCreated() {
-        if (getContainerId() == null) {
-            return false;
-        }
-
-        try {
-            String status = getCurrentContainerInfo().getState().getStatus();
-            return "created".equalsIgnoreCase(status) || isRunning();
-        } catch (DockerException e) {
-            return false;
-        }
+        return getContainerId() != null;
```

Comment:
@bsideup this makes more sense if we want to check that `start()` has been called at least once and to avoid NPEs.

It turns out that copying files into stopped containers also works fine.

Reply:
Please avoid changing existing behaviour, `getContainerId() != null` is not enough to check whether the container is created or not.

I would recommend relaxing the check in `copyFileFromContainer` instead of touching `isCreated`

Reply:
@bsideup I have to change the behaviour in some way for this to make sense.

There are situations where a container is *clearly created* but the existing logic for `isCreated()` will return `false`.

For example if a container is created and then stopped. It's status is `stopped` which matches neither `created` nor `isRunning()`.

This change doesn't cause any tests to fail so it would help me if you could explain in what situations it is not enough to check if `getContainerId() != null` - then we could perhaps try and improve this method.

Reply:
you absolutely don't have to change the `isCreated` method. Just remove the `isCreated()` check from `copyFileFromContainer` and replace with something more appropriate.

Reply:
`isCreated` method is there to check whether the container is in the created state or not *by querying the actual state*. `getContainerId() != null` would return true even if container is already exited long time ago, for example.

Reply:
> isCreated method is there to check whether the container is in the created state or not

but it returns true if the container is in the running state ?

Reply:
Correct.

Reply:
I am not clear what the "created state" in docker is. I guess you know it when you see it 🤷 .

Semantics aside, I have added a new method `exists()` and rebased.

Let me know what you think.

Reply:
Can we please avoid adding a new method and add a check to `copyFileFromContainer` instead?

Reply:
this is checked in `copyFileFromContainer` and `copyFileToContainer` so it really makes sense to extract a shared method.

Reply:
You _can_ extract a shared method, but we shouldn't be adding it as a member of our public API.

Reply:
hmm, but if you don't add it to the public API other implementations will no longer have a way to indicate to the `copyTo` / `copyFrom` methods that they are not in a state where that is supported.

it seems to me that the `isCreated` method doesn't mean what it looks like it means and that the public API is missing a way to check if a container exists. Which is why the bug exists in the first place.

Since this is an interface class there is not really a way to define a shared method in the class.

Reply:
other implementations are free to get the same mechanism (query the current container info, switch on the state field), so there are ways.

Also, `isCreated` is just a shortcut to `getCurrentContainerInfo().getState().getStatus()` + some `if`s, so saying that the public API is missing a way to check is a stretch :)

> Since this is an interface class there is not really a way to define a shared method in the class.

One way to do it is to define the method in an internal class.

Reply:
OK, for the record I think this is not the best solution but I've pushed a PR that puts this into a private method on an inner class

Reply:
FYI The isCreated method is only used by these copy methods - and it's the wrong check to do since. There aren't even any tests that exercise it.